src/Utilities/FlowAnalysis/FlowAnalysis/Framework/DataFlow/DataFlowOperationVisitor.cs

```diff
@@ -1260,29 +1261,41 @@ internal bool TryInferConversion(IIsPatternOperation operation, out ConversionIn
                 }
 
                 // Handle is pattern operations with constant pattern.
-                if (operation is IIsPatternOperation patternOperation &&
-                    patternOperation.Pattern is IConstantPatternOperation constantPattern)
+                if (operation is IIsPatternOperation isPatternOperation)
                 {
-                    if (constantPattern.Value.ConstantValue.HasValue)
+                    IPatternOperation patternOperation = isPatternOperation.Pattern;
+                    bool direct = true;
+
+                    if (INegatedPatternOperationWrapper.IsInstance(patternOperation))
+                    {
+                        INegatedPatternOperationWrapper negatedPattern = INegatedPatternOperationWrapper.FromOperation(patternOperation);
+                        patternOperation = negatedPattern.Pattern;
+                        direct = false;
+                    }
+
+                    if (patternOperation is IConstantPatternOperation constantPattern)
```

Comment:
```suggestion
                    if (patternOperation is IConstantPatternOperation { Value.ConstantValue: var constantValue )
```

It's unclear if the return statement could be refactored to let this reduce all the way to:

```suggestion
                    if (patternOperation is IConstantPatternOperation { Value.ConstantValue: { HasValue: true, Value: null } )
```

Reply:
I did the minimal amount of changes, not refactored the code.
The latter seems to work, but removes the `return false` case when the pattern is `IConstantPattern` but without a value. Not sure when that even would occur.
Pushed up a commit with that change.